### PR TITLE
use pypi.eclass

### PR DESCRIPTION
**Pull request by littlewu2508**

### CODE REVIEW

**Comment by heroxbd** [comment]
What are still needed to be done?

**Comment by heroxbd** [comment]
LGTM

**Comment by littlewu2508** [comment]
> What are still needed to be done?

Tests of downloading distfiles. I didn't perform the check, just did the automatic `sed` on ebuilds and updates the hash of ebuilds

**Comment by littlewu2508** [comment]
Running `ebuild *.ebuild fetch` on all ebuilds result in some errors:

[fetch.log](https://github.com/jiegec/gentoo-pypi-sci/files/12206439/fetch.log)

Running `ebuild *.ebuild digest` result in some distfile cannot be downloaded:
[digest.log](https://github.com/jiegec/gentoo-pypi-sci/files/12206442/digest.log)

**Comment by littlewu2508** [comment]
I believe there are package names not obeying the pypi.eclass' rules, so it gives some wrong SRC_URI

**Comment by littlewu2508** [comment]
I tried `pypi_sdist_url --no-normalize` to get things right but it didn't worked out -- it seemed that there's a bug in `pypi.eclass`. I'll investigate it and submit PR to fix it.

**Comment by heroxbd** [comment]
I think it is a good chance to ditch some of the packages like uproot3.

File: dev-python/cupy/cupy-12.1.0.ebuild [hunk 1]
```diff
@@ -33,15 +33,15 @@ DEPEND="
 		>=sci-libs/hipBLAS-${ROCM_VERSION}[${ROCM_USEDEP}]
 		>=sci-libs/hipCUB-${ROCM_VERSION}[${ROCM_USEDEP}]
 		>=sci-libs/hipFFT-${ROCM_VERSION}[${ROCM_USEDEP}]
-		>=sci-libs/rocRAND-${ROCM_VERSION}[${ROCM_USEDEP}]
+		>=sci-libs/hipRAND-${ROCM_VERSION}[${ROCM_USEDEP}]
```
Comment (littlewu2508) [inline dev-python/cupy/cupy-12.1.0.ebuild:36]: There's no hipRAND package in ::gentoo. It's rocRAND

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> There's no hipRAND package in ::gentoo. It's rocRAND

https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=d5326b32678303ec9ece262931f0b0e5588d4fc6

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> There's no hipRAND package in ::gentoo. It's rocRAND

https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=9edc7b68ee8a0e1f2e6e18f4675e97e5f257dc7f

**Comment by littlewu2508** [comment]
Now all packages has switched to `pypi.eclass` smoothly.

**Comment by heroxbd** [comment]
LGTM, again.

**Pull request merged**